### DISCUSSION TITLE
Query regarding applying voronoi cells to a polyhydron

### DISCUSSION THREAD

**Discussion by KALEEMULLAHJANKHAN** [neperfepx/neper#365] [Q&A] [answered]
1. I have a rock slope composed of different polyhedral blocks, Can I convert each of the polyhedral blocks to Voronoi cells using Neper?

2. Secondly, is there any command in Neper to import and generate the Voronoi tessellations of an existing structure other than a cube?

**Comment by rquey** [comment, marked as answer]
> 1. I have a rock slope composed of different polyhedral blocks, Can I convert each of the polyhedral blocks to Voronoi cells using Neper?

Probably. A tessellation can be generated from different types of input. It can be the number of cells (see `-n`), their centroids and sizes (see `-morpho`), etc.

> 2. Secondly, is there any command in Neper to import and generate the Voronoi tessellations of an existing structure other than a cube?

Somewhat. See `-domain`.

**Reply by KALEEMULLAHJANKHAN** [reply]
Thank you sir for your response. The answer is still not very clear to me. My question is that "I already have a polyhedral block from another source, and now I want to generate its voronoi tessellations in neper", How can I do it, I can't find any command for it.

**Reply by rquey** [reply]
"a polyhedral block from another source" is vague to me. Please provide more specifics or an illustration. See https://github.com/rquey/neper/discussions/95.

**Reply by KALEEMULLAHJANKHAN** [reply]
![Polyhedrons](https://user-images.githubusercontent.com/89472468/149964590-97dcd985-f9ce-4698-8f95-64be4472d206.JPG)
Dear sir, polyhedrons are attached to this reply. How can I apply voronoi tessellations to it in Neper?

**Reply by rquey** [reply]
You can generate a tessellation for each polyhedron independently.

The only way to obtain the desired polyhedral domain is by using option `-domain planes(...)`, as described in the documentation, where the planes are the bounding planes of the polyhedron - you have to know / specify their equations.
